### PR TITLE
Add a simple script to check that the policy is correct

### PR DESCRIPTION
**Pull request by vpodzime**
Can easily be used as a git hook (e.g. pre-commit).

### CODE REVIEW

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @nickanderson can review this?

File: tests/check_policy.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+fail=0
+if [ -n "$cf_promises" ]; then
+  if [ ! -f "promises.cf" ]; then
+    echo "Running autogen.sh to get the policy to verify"
+    ./autogen.sh
+  fi
+  echo "Verifying policy with cf-promises"
+  $cf_promises promises.cf || fail=1
+  $cf_promises update.cf   || fail=1
```
Comment (nickanderson) [inline tests/check_policy.sh]: May I suggest the addition of standalone_self_upgrade.cf as well? It's another entry (only used during self upgrade).

File: tests/check_policy.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#
+# Verify that the policy is correct. If you want to avoid commiting invalid
+# changes, just do:
+#   cp -a tests/check_policy.sh .git/hooks/pre-commit
+
+cf_promises=""
+if type cf-promises >/dev/null 2>&1; then
+  cf_promises="cf-promises";
+elif [ -x "../core/cf-promises/cf-promises" ]; then
+  cf_promises="../core/cf-promises/cf-promises"
```
Comment (nickanderson) [inline tests/check_policy.sh]: adding --full-check and --eval-functions gives a deeper check.

File: tests/check_policy.sh (hunk 1 above)
Comment (vpodzime) [reply tests/check_policy.sh]: Done.

File: tests/check_policy.sh (hunk 2 above)
Comment (vpodzime) [reply tests/check_policy.sh]: Nice! Added.

**Pull request merged**